astropy/coordinates/representation/base.py

```diff
@@ -0,0 +1,1600 @@
+from __future__ import annotations
+
+import abc
+import functools
+import inspect
+import operator
+import warnings
+
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates.angles import Angle
+from astropy.utils import ShapedLikeNDArray, classproperty
+from astropy.utils.data_info import MixinInfo
+from astropy.utils.exceptions import DuplicateRepresentationWarning
+
+__all__: list[str] = []  # Not public. Import from ``astropy.coordinates``
```

Comment:
I disagree with this still. Let's put things in here that get transferred up (like is done in modules in `builtin_frames`). If you feel strongly otherwise, I would suggest for now not having any `__all__` (like in `matrix_utilities`, which is also not exposed).  Though to me it is helpful to the code reader -- the ones for whom this matters most -- that it lists the classes, etc., that this file actually defines for use elsewhere.